routemaster/state_machine/types.py

```diff
@@ -1,8 +1,16 @@
 """Shared types for state machine execution."""
 
-from typing import Any, Dict, NamedTuple
+from typing import Any, Dict, Callable, NamedTuple
+
+from routemaster.app import App
+from routemaster.config import State, StateMachine
 
 Metadata = Dict[str, Any]
+IsExitingCheck = Callable[[], bool]
+StateProcessor = Callable[
+    [App, State, StateMachine, IsExitingCheck],
+    None,
+]
```

Comment:
At what point does it become more sensible to have a superclass for things like this with a well-defined interface rather than making this a callable? I am unsure whether we are past that point.